packages/react/src/NavList/NavList.tsx

```diff
@@ -413,7 +414,141 @@ const Group: React.FC<NavListGroupProps> = ({title, children, sx: sxProp = defau
   )
 }
 
-Group.displayName = 'NavList.Group'
+// ----------------------------------------------------------------------------
+// NavList.GroupExpand
+
+type GroupItems = {
+  text: string
+  trailingVisual?: Icon | string
+  leadingVisual?: Icon | string
+  trailingAction?: ActionListTrailingActionProps
+  'data-expand-focus-target'?: string
+} & Omit<NavListItemProps, 'children'>
+
+export type NavListGroupExpandProps = {
+  label?: string
+  pages?: number
+  items: GroupItems[]
+  renderItem?: (item: {text: string}) => React.ReactNode
+}
+
+export const GroupExpand = React.forwardRef<HTMLButtonElement, NavListGroupExpandProps>(
+  ({label = 'Show more', pages = 0, items, renderItem, ...props}, forwardedRef) => {
+    const [currentPage, setCurrentPage] = React.useState(0)
+    const groupId = useId()
+
+    const teamEnabled = useFeatureFlag('primer_react_css_modules_team')
+    const staffEnabled = useFeatureFlag('primer_react_css_modules_staff')
+
+    const {amountToShow, focusTargetIndex} = React.useMemo(() => {
+      const itemsPerPage = items.length / pages
+      const amountToShow = pages === 0 ? items.length : Math.ceil(itemsPerPage * currentPage)
+      const focusTargetIndex = currentPage === 1 ? 0 : amountToShow - Math.floor(itemsPerPage)
+
+      return {amountToShow, focusTargetIndex}
+    }, [pages, items, currentPage])
+
+    return (
+      <>
+        {currentPage > 0 ? (
+          <>
+            {items.map((itemArr, index) => {
+              const {
+                text,
+                trailingVisual: TrailingVisualIcon,
+                leadingVisual: LeadingVisualIcon,
+                trailingAction,
+                ...rest
+              } = itemArr
+              const {icon, label: actionLabel, ...actionProps} = trailingAction || {}
+              const focusTarget = index === focusTargetIndex ? groupId : 'false'
```

Comment:
```suggestion
              const focusTarget = index === focusTargetIndex ? groupId : undefined
```

Does it make sense to have the value be `undefined` so that the data attribute is not set or do we want it to be a false-y value intentionally?

Reply:
I think either way is fine! I'm thinking that if it's not set to the group ID, then we don't really care about that item. We could have the data attribute persist, but I'm not sure if there's much of a difference to how it is now? 🤔

Reply:
@TylerJDev totally, really this was just to omit the data attribute if this value is `false` since we're not going to use it. To me this can be helpful (basically don't add things to the DOM that we don't need to help with clarity) but totally get if they are equivalent or if it doesn't seem that helpful 👍

Reply:
AH, I had it confused. I assumed the `undefined` was how it was now. Not sure how I misread that 😆 Yeah, I think omitting the `data-*` is preferred!